drivers/soft_uart/soft_uart.c

```diff
@@ -17,7 +17,6 @@
  */
 
 #include <stdio.h>
-#include <assert.h>
```

Comment:
`drivers/soft_uart/include/soft_uart_params.h` now needs to include `kernel_defines.h` to get `ARRAY_SIZE` which was previously transitively included by `assert.h`

Reply:
Added in 0e9fee4.